streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -77,12 +81,13 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
         for (String topic : topics) {
             List<PartitionInfo> partitions = metadata.partitionsForTopic(topic);
 
-            if (partitions == null)
-                throw new StreamsException("Topic not found during partition assignment: " + topic);
-
-            int numPartitions = partitions.size();
-            if (numPartitions > maxNumPartitions)
-                maxNumPartitions = numPartitions;
+            if (partitions == null) {
+                log.warn("Topic not found during partition assignment: " + topic);
```

Comment:
Should we treat it as a WARN level entry? I think this situation can actually happen frequently in a real production cluster and is transient anyways, hence I'd suggest changing it to `log.info("Skipping assigning topic {} to tasks since its metadata is not available yet", topic);`.